### PR TITLE
Added preferred maximum-path configuration from broadcom

### PR DESCRIPTION
**Pull request by Cellebyte**
* Add suggestions from broadcom via Environment Variable.
* hotfix.

### CODE REVIEW

File: pkg/frr/configure.go [hunk 1]
```diff
@@ -91,15 +93,26 @@ func (f *FRRManager) renderSubtemplates(in FRRConfiguration) (*FRRTemplateConfig
 	if err != nil {
 		return nil, err
 	}
+	envECMPMaximumPaths, isSet := os.LookupEnv("NWOP_ECMP_MAXIMUM_PATHS")
```
Comment (schrej) [inline pkg/frr/configure.go:96]: FYI: The second return value is commonly called `ok` for things like this, but no need to change that.

File: pkg/frr/configure.go [hunk 2]
```diff
@@ -91,15 +93,26 @@ func (f *FRRManager) renderSubtemplates(in FRRConfiguration) (*FRRTemplateConfig
...
+	ecmpMaximumPaths := 8 // Default to BCMs preferred target of maximum 8 paths per route.
+	if isSet {
+		parsedECMPMaximumPaths, err := strconv.Atoi(envECMPMaximumPaths)
+		if err != nil {
+			ecmpMaximumPaths = parsedECMPMaximumPaths
+		} else {
+			return nil, err
+		}
+	}
```
Comment (schrej) [inline pkg/frr/configure.go:105]: I would implement it like this, which is a more common style and reduces scope of some of the variables.

In addition the if condition for error handling was the wrong way round, which is corrected here as well.

```suggestion
	ecmpMaximumPaths := 8 // Default to BCMs preferred target of maximum 8 paths per route.
	if envECMPMaximumPaths, ok := os.LookupEnv("NWOP_ECMP_MAXIMUM_PATHS"); ok {
		parsedECMPMaximumPaths, err := strconv.Atoi(envECMPMaximumPaths)
		if err != nil {
			return nil, err
		}
		ecmpMaximumPaths = parsedECMPMaximumPaths
	}
```

**Review by schrej — CHANGES_REQUESTED** [review]
One bug, lgtm otherwise

**Review by schrej — APPROVED** [review]
go part lgtm

**Comment by Cellebyte** [comment]
should we still add this feature as a global configuration or not?

**Comment by MaxRink** [comment]
@Cellebyte we should default to something sane

**Comment by Cellebyte** [comment]
> @Cellebyte we should default to something sane

What is something sane @MaxRink ?

**Comment by Cellebyte** [comment]
Closed as not needed

**Pull request closed without merging**